src/helper/service.cpp

```diff
@@ -72,6 +77,36 @@ service::ptr service::from_settings(const service_identifier &id,
         std::chrono::milliseconds{rc_settings.poll_interval});
 }
 
+void service::handle_error()
+{
+    rc_action_ = [this] { discover(); };
+    errors_++;
+    interval_ = std::chrono::duration_cast<std::chrono::milliseconds>(
+        poll_interval_ * pow(2, std::min((errors_ - 1), max_increment)));
+}
+
+void service::poll()
+{
+    try {
+        rc_client_->poll();
+    } catch (dds::remote_config::network_exception & /** e */) {
+        handle_error();
+    }
+}
+void service::discover()
+{
+    try {
+        if (rc_client_->is_remote_config_available()) {
+            // Remote config is available. Start polls
+            rc_action_ = [this] { poll(); };
+            errors_ = 0;
+            interval_ = poll_interval_;
+            return;
```

Comment:
Nitpick, but I would remove this return from here and just call `handle_error` inside the catch body

Reply:
Ok

Reply:
This is done this way so `handle_error` is called whenever rc is not enabled